core/src/main/scala/akka/kafka/internal/CommitCollectorStage.scala

```diff
@@ -79,18 +91,24 @@ private final class CommitCollectorStageLogic(
     stage.in,
     new InHandler {
       def onPush(): Unit = {
+        var pushed = false
         val offset = grab(stage.in)
         log.debug("Consuming offset {}", offset)
         if (updateBatch(offset)) {
           // Push only of the outlet is available, a commit on interval might have taken the pending demand.
           // This is very hard to get tested consistently, so it gets this big comment instead.
-          if (isAvailable(stage.out)) pushDownStream(BatchSize)
+          if (isAvailable(stage.out)) {
+            pushDownStream(BatchSize)
+            pushed = true
+          }
         } else tryPull(stage.in) // accumulating the batch
+        if (!pushed) suspendContext()
```

Comment:
minor suggestion: 

Since here we have only one else branch to care about, we could have: 

```suggestion
          if (isAvailable(stage.out)) pushDownStream(BatchSize)
        } else {
          tryPull(stage.in) // accumulating the batch
          suspendContext()
        }
```

That removes the need for `var pushed = false` above.

Reply:
@octonato I think your code should be:

```
        if (updateBatch(offset)) {
          if (isAvailable(stage.out)) pushDownStream(BatchSize)
          else suspendContext()
        } else {
          tryPull(stage.in) // accumulating the batch
          suspendContext()
        }
```

then it's not much simpler, is it?

Reply:
I was trying to avoid repeating suspendContext in each branch other than those calling pushDownStream. I think this way it's less probable to forget to call suspendContext.